static/src/javascripts/lib/cmp.spec.js

```diff
@@ -0,0 +1,215 @@
+// @flow
+import { init, onConsentNotification, consentState, _ } from 'lib/cmp';
+import { getAdConsentState as _getAdConsentState } from 'common/modules/commercial/ad-prefs.lib';
+
+const getAdConsentState: any = _getAdConsentState;
+
+jest.mock('common/modules/commercial/ad-prefs.lib', () => ({
+    getAdConsentState: jest.fn(),
+    thirdPartyTrackingAdConsent: jest.fn(),
+}));
+
+describe('cmp', () => {
+    beforeEach(() => {
+        _.resetCmp();
+        getAdConsentState.mockReset();
+    });
+
+    it('does not re-trigger consent notifications if init called multiple times', () => {
+        const myCallBack = jest.fn();
+
+        init();
+
+        onConsentNotification('functional', myCallBack);
+
+        expect(myCallBack).toHaveBeenCalledTimes(1);
+
+        init();
+
+        expect(myCallBack).toHaveBeenCalledTimes(1);
+    });
+
+    describe('consentState', () => {
+        it('returns null state if cmp lib has not been initialised', () => {
+            expect(consentState('functional')).toBeNull();
+            expect(consentState('performance')).toBeNull();
+            expect(consentState('advertisement')).toBeNull();
+        });
+
+        it('returns functional consent state', () => {
+            init();
+
+            expect(consentState('functional')).toBe(true);
+        });
+
+        it('returns performance consent state', () => {
+            init();
+
+            expect(consentState('performance')).toBe(true);
+        });
+
+        it('returns advertisement consent state with true if getAdConsentState true', () => {
+            getAdConsentState.mockReturnValue(true);
+
+            init();
+
+            expect(consentState('advertisement')).toBe(true);
+        });
+
+        it('returns advertisement consent state with false if getAdConsentState true', () => {
+            getAdConsentState.mockReturnValue(false);
+
+            init();
+
+            expect(consentState('advertisement')).toBe(false);
+        });
+    });
+
+    describe('onConsentNotification', () => {
```

Comment:
Do we want to add tests to make sure a callback that gets registered prior to `init()`'s invocation are still executed, or does that not make sense from a "dependency of code" point of view? If the latter then no worries!

Reply:
I think that's a great suggested test!

Reply:
Turns out we covered that scenario already: https://github.com/guardian/frontend/blob/c4c4ac43e874985a25014e04a85a339a03c3dad7/static/src/javascripts/lib/cmp.spec.js#L137